appinventor/buildserver/src/com/google/appinventor/buildserver/resources/runtime.scm

```diff
@@ -407,7 +408,8 @@
            ;; for event handlers and component names.
            (let ((registeredObject (string->symbol registeredComponentName)))
                  (if (is-bound-in-form-environment registeredObject)
-                     (if (eq? (lookup-in-form-environment registeredObject) componentObject)
+                     ;; temporarily deleted because I have no ideas how to get true out of this condition using isMulti
+                     ;;(if (or (eq? (lookup-in-form-environment registeredObject) componentObject) (isMulti))
```

Comment:
@mark-friedman 
So previously the condition here is
(if (eq? (lookup-in-form-environment registeredObject) componentObject)

I want this condition to be something like this:
(if (or (eq? (lookup-in-form-environment registeredObject) componentObject) (isMulti))

where is isMulti is a boolean passed into the function. However, something about this is not working correctly, and I haven't been able to figure it out. I was wondering if you have an idea of what the issue is.

Also, the reason why this will be okay is because multi-generic events will be registered to some arbitrary instance of the right type. So when the event is dispatched, the registered object and the one that we pass in won't always match. However, it doesn't matter, because the two objects are of the same type (will be guaranteed by the event name, which will be of the form MultiGeneric-TypeName-EventName). Since, the multi-block handles events for all components of that that type, that is enough to know.